targets/target-mbed/source/jerry_port.c

```diff
@@ -0,0 +1,141 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define _BSD_SOURCE
+#include <stdarg.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+#include "jerry-core/jerry-port.h"
+
+#include "mbed-hal/us_ticker_api.h"
+
+static bool abort_on_fail = false;
+
+/**
+ * Provide log message to filestream implementation for the engine.
+ */
+int jerry_port_logmsg (FILE *stream, /**< stream pointer */
+                       const char *format, /**< format string */
+                       ...) /**< parameters */
+{
+  va_list args;
+  int count;
+  va_start (args, format);
+  count = vfprintf (stream, format, args);
+  va_end (args);
+  return count;
+} /* jerry_port_logmsg */
+
+/**
+ * Provide error message to console implementation for the engine.
+ */
+int jerry_port_errormsg (const char *format, /**< format string */
+                         ...) /**< parameters */
+{
+  va_list args;
+  int count;
+  va_start (args, format);
+  count = vfprintf (stderr, format, args);
+  va_end (args);
+  return count;
+} /* jerry_port_errormsg */
+
+/**
+ * Provide output character to console implementation for the engine.
+ */
+int jerry_port_putchar (int c) /**< character to put */
+{
+  return putchar ((unsigned char) c);
+} /* jerry_port_putchar */
+
+/**
+ * Sets whether 'abort' should be called instead of 'exit' upon exiting with
+ * non-zero exit code in the default implementation of jerry_port_fatal.
+ */
+void jerry_port_default_set_abort_on_fail (bool flag) /**< new value of 'abort on fail' flag */
+{
+  abort_on_fail = flag;
+} /* jerry_port_default_set_abort_on_fail */
+
+/**
+ * Check whether 'abort' should be called instead of 'exit' upon exiting with
+ * non-zero exit code in the default implementation of jerry_port_fatal.
+ *
+ * @return true - if 'abort on fail' flag is set,
+ *         false - otherwise.
+ */
+bool jerry_port_default_is_abort_on_fail ()
+{
+  return abort_on_fail;
+} /* jerry_port_default_is_abort_on_fail */
+
+/**
+ * Default implementation of jerry_port_fatal.
+ */
+void jerry_port_fatal (jerry_fatal_code_t code)
+{
+  if (code != 0
+      && code != ERR_OUT_OF_MEMORY
+      && jerry_port_default_is_abort_on_fail ())
+  {
+    abort ();
+  }
+  else
+  {
+    exit (code);
+  }
+} /* jerry_port_fatal */
+
+/**
+ * Implementation of jerry_port_get_time_zone.
+ */
+bool jerry_port_get_time_zone (jerry_time_zone_t *tz_p)
+{
+  struct timeval tv;
+  struct timezone tz;
+
+  /* gettimeofday may not fill tz, so zero-initializing */
+  
+  tz.tz_minuteswest = 0;
+  tz.tz_dsttime = 0;
+
+  gettimeofday (&tv, &tz);
```

Comment:
I'm not familiar with mbed's libc implementation, but from `_gettimeofday` (just some lines below) I guess that `gettimeofday` is just a trampoline to `_gettimeofday`. And that function does not do anything with the time zone parameter. So, I guess you'll have no info about TZ at all. If my hypothesis is right, then this function could be just:
```c
tz_p->offset = 0;
tz_p->daylight_saving_time = 0;
return true;
```
Without any calls to `gettimeofday`.